types/externs.d.ts

```diff
@@ -29,8 +53,16 @@ declare global {
   /** Make optional all properties on T and any properties on object properties of T. */
   type RecursivePartial<T> = {
     [P in keyof T]+?:
+      // If type is a union, map each individual component and transform the resultant tuple back into a union.
```

Comment:
![image](https://user-images.githubusercontent.com/4071474/72196752-3b0a5080-33cf-11ea-9bf4-b27131037075.png)

Reply:
This was needed for types such as:

```ts
export interface Json {
    passes: PassJson[] | null;
}
```

It would otherwise not recuse into `RecursivePartial<PassJson>`, as `PassJson[] | null` fails all the conditional checks and just returns itself (`T[P]`).|

Note, if the property is optional (and `LH.Config.Json['passes']` actually is), `SplitTypes<T>` turns it into `[undefined, PassJson[], null]`, so no special handling needed for that.

Reply:
Can you add a comment in the code with this concrete example? This helped a lot :)